processor/deltatocumulativeprocessor/reorderprocessor/internal/gmetric/iter.go

```diff
@@ -0,0 +1,60 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gmetric // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/reorderprocessor/internal/gmetric"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics/identity"
+)
+
+type List[T any] interface {
+	At(int) T
+	Len() int
+}
+
+func All[L List[T], T any](list L) func(func(int, T) bool) {
+	return func(yield func(int, T) bool) {
+		for i := 0; i < list.Len(); i++ {
+			if !yield(i, list.At(i)) {
+				break
+			}
+		}
+	}
+}
+
+type ResourceMetric struct {
```

Comment:
I think we should rename this to be less similar to ResourceMetrics.

Perhaps `MetricMetadata`?